File: src/main/java/core/procedures/strings/StartsWith.java
package core.procedures.strings;

import core.procedures.AFn;
import core.procedures.FnArgsBuilder;

public final class StartsWith extends AFn {

  public StartsWith() {
    super(new FnArgsBuilder().minArgs(2).maxArgs(2).mandatoryArgsTypes(new Class[]{CharSequence.class, CharSequence.class}));
  }

  @Override
  public String getName() {
    return "starts-with?";
  }

  @Override
  public Boolean apply2(Object arg1, Object arg2) {
    return arg1.toString().startsWith(arg2.toString());
  }
}

File: src/main/java/core/procedures/strings/StartsWith.kt
package core.procedures.strings

import core.procedures.AFn
import core.procedures.Arity.Exactly

class StartsWith : AFn<CharSequence?, Boolean>(name = "starts-with?", isPure = true, arity = Exactly(2),
                                               mandatoryArgsTypes = arrayOf(CharSequence::class.java,
                                                                            CharSequence::class.java)) {

    override operator fun invoke(arg1: CharSequence?, arg2: CharSequence?) = arg1!!.startsWith(arg2!!)
}
